packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -778,7 +795,7 @@ export class ModularChangeFamily
 			crossFieldKeys,
 			maxId: genId.getMaxId(),
 			revisions: revInfos,
-			constraintViolationCount: change.change.constraintViolationCount,
+			inputConstraintViolationCount: change.change.inputConstraintViolationCount,
```

Comment:
Don't we need to swap the input and output violation counts here.
Before fixing this, it would be good to write a test that detects the bug. I think you'll need to write a test where a transaction puts an undo constraints on the existence of a node that is removed on some other branch, then the transaction with the undo constraint should be rebased over the commits that removes the node (so that the output constraint is violated). Then the transaction should be undone, and that should have no effect.

Reply:
Good catch. Added test and fixed the issue.